azurerm/resource_arm_data_factory_linked_service_sql_server.go

```diff
@@ -0,0 +1,357 @@
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, dataFactoryName, name, "")
+	if err != nil {
+		return fmt.Errorf("Error retrieving Data Factory Linked Service SQL Server %q (Data Factory %q / Resource Group %q): %+v", name, dataFactoryName, resourceGroup, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Data Factory Linked Service SQL Server %q (Data Factory %q / Resource Group %q): %+v", name, dataFactoryName, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmDataFactoryLinkedServiceSQLServerRead(d, meta)
+}
+
+func resourceArmDataFactoryLinkedServiceSQLServerRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).dataFactoryLinkedServiceClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	dataFactoryName := id.Path["factories"]
+	name := id.Path["linkedservices"]
+
+	resp, err := client.Get(ctx, resourceGroup, dataFactoryName, name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error retrieving Data Factory Linked Service SQL Server %q (Data Factory %q / Resource Group %q): %+v", name, dataFactoryName, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("data_factory_name", dataFactoryName)
+
+	sqlServer, ok := resp.Properties.AsSQLServerLinkedService()
+	if !ok {
+		return fmt.Errorf("Error classifiying Data Factory Linked Service SQL Server %q (Data Factory %q / Resource Group %q): Expected: %q Received: %q", name, dataFactoryName, resourceGroup, datafactory.TypeSQLServer, *resp.Type)
+	}
+
+	d.Set("additional_properties", sqlServer.AdditionalProperties)
+
+	if sqlServer.Description != nil {
+		d.Set("description", *sqlServer.Description)
+	}
+
+	annotations := flattenDataFactoryLinkedServiceAnnotations(sqlServer.Annotations)
+	if err := d.Set("annotations", annotations); err != nil {
+		return fmt.Errorf("Error setting `annotations`: %+v", err)
+	}
+
+	parameters := flattenDataFactoryLinkedServiceParameters(sqlServer.Parameters)
+	if err := d.Set("parameters", parameters); err != nil {
+		return fmt.Errorf("Error setting `parameters`: %+v", err)
+	}
+
+	if connectVia := sqlServer.ConnectVia; connectVia != nil {
+		if connectVia.ReferenceName != nil {
+			d.Set("integration_runtime_name", *connectVia.ReferenceName)
+		}
+	}
+
+	if properties := sqlServer.SQLServerLinkedServiceTypeProperties; properties != nil {
+		if properties.ConnectionString != nil {
+			val, ok := properties.ConnectionString.(string)
+			if !ok {
+				log.Printf("[DEBUG] Skipping connection string %q since it's not a string", val)
+			} else {
+				d.Set("connection_string", properties.ConnectionString.(string))
+			}
+		}
+	}
```

Comment:
I think it'd be better to make this:

```
connectionString := ""
if properties := sqlServer.SQLServerLinkedServiceTypeProperties; properties != nil {
    if properties.ConnectionString != nil {
    val, ok := properties.ConnectionString.(string)
    if ok {
      connectionString = val
    } else {
      log.Printf("[DEBUG] Skipping connection string %q since it's not a string", val)
    }
  }
}

d.Set("connection_string", connectionString)
```